portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,10 +33,11 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> comments = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<String> comments = getDatastoreComments();
+
     String json = "{\"comments\": [";
     for (String comment : comments)
```

Comment:
Optional: you could make this a bit more concise by eliminating the `comments` and just calling `getDatastoreComments()` directly in the target of the for-each loop.